### ISSUE TITLE
Messaging -  One Sentence Summary for Kedro

### ISSUE THREAD

**Issue by NeroOkwa** [kedro-org/kedro-devrel#2]

**Comment by stichbury** [comment]
ChatGPT came up with this "Our mission is to standardise how data science code is created" while I was playing around recently.

**Comment by yetudada** [comment]
Another one "production data science"

**Comment by astrojuanlu** [comment]
Looking at Django and thinking:

![image](https://user-images.githubusercontent.com/316517/219765590-6563ad19-c8fd-40a1-9c59-09b90e06c18b.png)

On one hand, there's a slogan or catchphrase. "Clean code for data scientists" and "Production data science" and "Think maintainable datascience code" are good slogans. Good for H1 on the website, marketing, etc.

And on the other hand, there's a one sentence summary. "An open source Python framework to create reproducible, maintainable, and modular data science code" is descriptive and succint, and so I think it is a good summary. A shorter variation: "An open source Python framework for modular data science code". Another idea: "An ecosystem/constellation of Python libraries for agile data science" (ecosystem includes extensions like kedro-viz and others, "agile" might be too overloaded at this point but wanted to throw it anyway).

I think a slogan has to be more evocative and colorful, while a summary or description has to be more conventional.

**Comment by yetudada** [comment]
I'm building on thoughts dropped by @astrojuanlu. We need to support a few things: a slogan, category and short description for technical and non-technical users. 

I would like to see if we can vote on five one-liners per category and get users to vote on them. 

## Slogan suggestions 
> Current: Maintainable and modular data science solved

- A clean code toolbox for machine learning
- A toolbox for production-ready data science 
- The software engineering toolbox for data scientists
- Automate machine-learning engineering plumbing 
- The missing link between machine learning and software engineering
- Machine Learning + Software Engineering = Kedro
- Think maintainable data science
- Put some software engineering back into machine-learning code
- Reduce technical debt in machine-learning code with Kedro
- Scaffolding for data scientists to create maintainable code
- Software engineering plumbing for successful machine-learning prototypes
- Write production-ready machine-learning code from the beginning
- Make machine-learning code that looks like Lego

## Category suggestions 
> Current: "Orchestrator" (incorrect), "Workflow", "AI Framework & Libraries" and "MLOps". Additional complexity, the word "framework" has been co-opted by tools like Hugging Face, PyTorch and Tensorflow. 

- MLE (machine-learning engineering)
- MLE frameworks
- MLOPs workflow
- Pipeline framework
- ML workflow 
- ML development framework
- ML code development tools

## One-sentence summary for Kedro (technical audience)

> Current: A Python framework for creating reproducible, maintainable and modular data science code.

- An open source Python framework for modular data science code.
- An open source Python framework that combines software engineering and machine learning to create code you can trust and maintain.
- An open source Python toolbox that applies software engineering concepts to machine-learning code so that teams can reduce the technical debt of their machine-learning prototypes.
- An open source Python toolbox that puts software engineering back into machine-learning code, making it easier to take prototypes into production systems. 

## One-sentence summary for Kedro (non-technical audience)

> Current: A Python framework for creating reproducible, maintainable and modular data science code.

- Kedro is the skeleton of your ML projects, much like physical scaffolding when constructing a building.
- While regular machine-learning codebases look more like sandcastles, Kedro encourages users to create castles with Lego. If one part starts crumbling, it's easier to repair and replace. In addition, each lego block follows a uniform format across all Kedro projects - making it easier to reuse. With this metaphor, Kedro projects are more robust against time because of structure, organisation and the design of their Lego blocks.

**Comment by stichbury** [comment]
Just a few notes/thoughts/ideas

> ## Slogan suggestions
> * A clean code toolbox for machine learning
I think this would need to be hyphenated to "**_A clean-code toolbox for machine learning_**" to avoid ambiguity

> * Automate machine-learning engineering plumbing
Very hard to parse. But I can't think of much improvement except "**_Plumb in the engineering for your machine-learning project. Automatically._**" 😱  

> * Write production-ready machine-learning code from the beginning
"**_Production-ready machine-learning code from the start_**" ? "**_Production-ready machine-learning code from the outset_**" ?

> * Make machine-learning code that looks like Lego
Probably avoid this -- it's likely trademarked

> ## Category suggestions

> ## One-sentence summary for Kedro (technical audience)

> * An open source Python framework that combines software engineering and machine learning to create code you can <s>trust and</s> maintain.

You're writing most of the code still, and trust is 🤷 so maybe a shorter version is "**_An open source Python framework that combines software engineering and machine learning to create maintainable code_**"

> * An open source Python toolbox that applies software engineering concepts to machine-learning code so that teams can reduce the technical debt of their machine-learning prototypes.

I'd argue that Kedro doesn't reduce technical debt -- if you have tech debt already, Kedro isn't helpful. It reduces the chances of creating technical debt in future projects.

Couple of suggestions:

* "**_Kedro uses software engineering best practices to help you build data science code ready for production_**". 
* "**_Our mission is to standardise how data science code is created to help you build maintainable production-ready projects_**"

With both of the above I've gone for using "you" and "our" to make it less formal.

> ## One-sentence summary for Kedro (non-technical audience)

> * Kedro is the skeleton of your ML projects, much like physical scaffolding when constructing a building.

* "**_Kedro is the backbone of your ML projects; it's like the scaffolding used in construction_**". ?

**Comment by astrojuanlu** [comment]
Given that the directory structure, guardrails, and decisions that Kedro makes on behalf of users seem to be an important principle, do we want to include the word "opinionated" or synonyms? xref https://github.com/kedro-org/kedro/issues/2388

Examples:

> Black, the uncompromising code formatter
> Black is the uncompromising Python code formatter. By using it, you agree to cede control over minutiae of hand-formatting. In return, Black gives you speed, determinism, and freedom from pycodestyle nagging about formatting.

https://pypi.org/project/black/

> Opinionated lightweight ELT pipeline framework
> A lightweight opinionated ETL framework, halfway between plain scripts and Apache Airflow 

https://pypi.org/project/mara-pipelines/

**Comment by yetudada** [comment]
Here are the results.

## Results

### Slogan 

Votes:

- Maintainable and modular data science solved (22 🗳️)
- A toolbox for production-ready machine learning (22 🗳️)
- The software engineering toolbox for data scientists (20 🗳️)

Comments of interest:

- I'd add 'Production ready data science with ease' and would argue to not use the word 'machine learning' in any case as there are many data wrangling/engineering for analytics tasks/pipelines which are no ML.
- I won't refer to kedro as a "toolbox" since it is very opinionated : it is not a stack of utilities you can cherry pick.
- I think kedro is much more capable than for machine learning. It also handles the heavy data prep work that can underly data workloads that calculate aggregates and apply business logic. So with that, it brings engineering principles to data scientists that might be used to basically MS Excel but in python. It takes the spaghetti out of those more fundamental parts of the data science hierarchy of needs. Second choice would be the first option, but it assumes that the audience knows that maintainable, modular data science code is a problem. Any academic will tell you their code works fine, until they have to run it on somebody else’s computer, or run sensitivity experiments. Not everyone has the scar tissue awareness of collaborative data science.

Final suggestion:

- Stick with "Maintainable and modular data science solved" and use "A toolbox for production-ready machine learning" in the one-sentence summary of Kedro; "A toolbox for production-ready machine learning" should be "A toolbox for production-ready data science" instead

## Category

Votes:

- ML development framework (35 🗳️)
- Machine learning engineering framework (25 🗳️)
- MLOPs framework (4 🗳️)

Comments of interest: 

- Kedro is definitely focused on "development" but not really on the "ops" side so I'd like this option better. _(ML development framework)_
- It is not covering a large part of ML ops. It is one of the building blocks (you still need AWS / airflow etc). ML engineering is better than ML development (job connotation better). _(Machine learning engineering framework)_
- It's not a MLOps framework (no monitoring, no deployment), it's a DS/ML development framework. _(ML development framework)_
- Kedro is not an MLOps framework; at best, it's part of an MLOps ecosystem. You'd expect an MLOps framework to help you with deployment, feature store, model serving, monitoring, etc.

Final suggestion:
 - "ML development framework" but it might need to be "DS development framework" 

## Describing Kedro to a technical user

Votes:

- An opinionated Python framework for creating reproducible, maintainable and modular machine-learning code. (39 🗳️)
- An opinionated Python framework that applies software engineering principles to machinelearning code. (25 🗳️)

Comments of interest:

- Replace ML with DS
- For me setting up a templated package with tests, makefile, pyproject and /src is not the big deal. I was doing that already on my own. But helping me keep code more maintainable due to organizing it in nodes/ pipelines and the vizualization is the big win for kedro. This is not about software engineering principles.

Final result:

- "An opinionated Python framework for creating reproducible, maintainable and modular machine-learning code." is the winner; it becomes "An opinionated Python framework for creating reproducible, maintainable and modular data science code."
- Additionally, we can extend this definition to be "An opinionated Python framework for creating reproducible, maintainable and modular data science code. It reduces technical debt when moving prototypes into production." and this absorbs some of the language in the definition for a non-technical user

## Describing Kedro to a non-technical user

Votes:

- A Python toolbox that puts software engineering back into machine-learning code, making it easier to take prototypes into production systems. (35 🗳️)
- The backbone of your ML prototypes; it's like the scaffolding used in construction. (16 🗳️)
- A Python toolbox that applies software engineering concepts to machine-learning code so that teams can reduce the technical debt of their machine-learning prototypes. (13 🗳️)

Comments of interest:

- Id think that structure, clarity and reusability of a kedro'd codebase will be much better than without. replace ml with ds. 'puts back' sounds somewhat strange. what about 'enforces better coding practices'
- I'd definitely emphasizes on the transition to production if I talk to a non technical PM. When talking to an engineering manager, I'd more likely emphasizes on the first sentence about "reducing technical debt"
- Only change would be that software engineering is not going “back into” machine learning code. I would argue that it was never there in the first place (see kaggle).
- Between first two: A python toolbox that applies software engineering concepts to machine-learning code so that teams can reduce technical debt when switching from prototypes to production. The points are: - reduce tech debt - go from prototype to production - Kedro is **not** helping in prototyping! It actually makes it slower!
- The point is we want to avoid the gap between prototypes and production. If you only focus on the prototyping/development, I think people will not see the added value over the flexibility of a notebook
- I would refer target Data Science in general, not only Machine Learning, as there exists many data engineering pipelines that could benefit of kedro without applying ML models
- not sure a non technical product manager will really figure out technical debt. - why not mention it covers from exploratory to production?

Final result:
- "A Python toolbox that puts software engineering back into machine-learning code, making it easier to take prototypes into production systems." won but it will be changed to "A Python toolbox that applies software engineering principles on data science code, making it easier to transition from prototype to production."
